python/cuspatial/cuspatial/core/binops/binops.py

```diff
@@ -0,0 +1,409 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+from abc import ABC, abstractmethod
+
+import cudf
+
+from cuspatial.core._column.geocolumn import GeoColumn
+from cuspatial.core.binops.contains import contains_properly
+from cuspatial.utils.column_utils import (
+    contains_only_linestrings,
+    contains_only_multipoints,
+    contains_only_points,
+    contains_only_polygons,
+    has_same_dimension,
+)
+
+
+class _binop(ABC):
```

Comment:
Rename this to `_BinaryPredicate`?

Reply:
Agree. binop typically means arithmetic operations.